Pipeline ID help text

MatthewHambley opened the pull request:
As per issue #10 - Provides help on IDs to use when specifying pipelines.

The original plan was to provide a separate argument which would spit out the list but `argparse` does not easily support that mode of operation. Instead they are always printed out with the help text. This is probably a better solution while the number is small but might become unwieldy later.

matthewrmshin commented on source/stylist/__main__.py:
```diff
@@ -27,14 +27,29 @@ def parse_cli() -> argparse.Namespace:
+    parsers = [key + ' ' * (8 - len(key)) + '- ' + str(parser)
+               for key, parser in _LANGUAGE_MAP.items()]
+    preproc = [key + ' ' * (8 - len(key)) + '- ' + str(proc)
+               for key, proc in _PREPROCESSOR_MAP.items()]
```
You may want to try [str.ljust](https://docs.python.org/3/library/stdtypes.html#str.ljust)?

matthewrmshin replied:
I guess this does not stop a language having multiple pre-processors?

MatthewHambley replied:
A good recommendation, `str.ljust` cleans up the code quite a bit. While I was at it I made the padding dynamic based on the content rather than fixed.

And yes, an extension pipeline can have multiple preprocessors as the pFUnit one does. It has to first go through the pFUnit processor, then the standard Fortran processor.

MatthewHambley commented:
It's hard to see how we would unit test the command line parsing. I imagine it would involve a lot of messing about and general unpleasantness. However I have added tests for the names I added to the source classes.

matthewrmshin commented:
> It's hard to see how we would unit test the command line parsing. I imagine it would involve a lot of messing about and general unpleasantness.

Yes, it will be a bit messy, but it should be possible to write command line parsing tests using mock, [monkeypatch](https://docs.pytest.org/en/latest/monkeypatch.html), etc.

But perhaps in a separate pull request.
